config/routes.rb

```diff
@@ -0,0 +1,7 @@
+Rails.application.routes.draw do
+  get '/users', to: 'users#index'
+  get '/users/:id', to: 'users#show'
+  get '/users/:id/posts', to: 'posts#index'
+  get '/users/:id/posts/:id', to: 'posts#show'
```

Comment:
The Rails router recognizes URLs and dispatches them to a controller's action, or to a Rack application. It can also generate paths and URLs, avoiding the need to hardcode strings in your views. However, it would be a better practice to use the default Resource routing.
Resource routing allows you to quickly declare all of the common routes for a given resourceful controller. A single call to resources can declare all of the necessary routes for your index, show, new, edit, create, update, and destroy actions.
You can do something of this sort `resources :posts, only: [:index, :show]` and this will only connect the posts index and show methods. Do same thing for users index and show method.

Reply:
This issue is well resolved! ✔️